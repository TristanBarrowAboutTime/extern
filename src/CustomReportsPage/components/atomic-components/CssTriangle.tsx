import * as React from 'react';
import styled from 'styled-components';

type StyleProps = {
    size: number,
    color: string,
}

const defaultColor = '#000000';
const defaultSize = 5;

const UpArrow = styled.div`
    width: 0;
    height: 0;
    border-left: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-right: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-bottom: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

const DownArrow = styled.div`
    width: 0;
    height: 0;
    border-left: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-right: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-top: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

const LeftArrow = styled.div`
    width: 0;
    height: 0;
    border-top: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-bottom: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-right: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

const RightArrow = styled.div`
    width: 0;
    height: 0;
    border-top: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-bottom: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-left: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

type CssTriangleProps = {
    style?: {}
    onClick?: () => void
    color?: string
    size?: number
}

export const Up = ({
    style = {},
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div style={style} onClick={onClick}>
            <UpArrow size={size} color={color} />
        </div>
    );
}

export const Down = ({
    style = {},
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div style={style} onClick={onClick}>
            <DownArrow size={size} color={color} />
        </div>
    );
}

export const Left = ({
    style = {},
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div style={style} onClick={onClick}>
            <LeftArrow size={size} color={color} />
        </div>
    );
}

export const Right = ({
    style = {},
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div style={style} onClick={onClick}>
            <RightArrow size={size} color={color} />
        </div>
    );
}

export const VSpacer = ({size}: {size: number}) => {
    return (
        <div style={{
            width: 1,
            height: size
        }}/>
    );
}

export const HSpacer = ({size}: {size: number}) => {
    return (
        <div style={{
            width: size,
            height: 1
        }}/>
    );
}