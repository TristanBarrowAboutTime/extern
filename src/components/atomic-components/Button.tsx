import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import Styles from '../../style/Styles';

/**
 * This will probably need to be turned into react-native at some point.
 */
export enum ButtonType {
    SMALL_NORMAL = 'SMALL_NORMAL',
    SMALL_GREEN = 'SMALL_GREEN',
    SMALL_BARE = 'SMALL_BARE',
    SMALL_RED = 'SMALL_RED',
    NORMAL = 'NORMAL',
    GREEN = 'GREEN',
    BARE = 'BARE',
    RED = 'RED',
}

type StyledButtonProps = {
    buttonType: ButtonType
    disabled: boolean
}

const getBgColor = (props: StyledButtonProps): string => {
    if (props.disabled) {
        switch(props.buttonType) {
            case ButtonType.NORMAL: 
            case ButtonType.SMALL_NORMAL:
            case ButtonType.GREEN:  
            case ButtonType.SMALL_GREEN:
            case ButtonType.RED:
            case ButtonType.SMALL_RED:
                return Styles.color.gray.medium;
            case ButtonType.BARE:
            case ButtonType.SMALL_BARE:
                return Styles.color.transparent;
            default:
                return Styles.color.gray.medium;
        }
    } else {
        switch(props.buttonType) {
            case ButtonType.NORMAL: 
            case ButtonType.SMALL_NORMAL:
                return Styles.color.gray.xx_dark;
            case ButtonType.GREEN:  
            case ButtonType.SMALL_GREEN:
                return Styles.color.green;
            case ButtonType.RED:
            case ButtonType.SMALL_RED:
                return Styles.color.red;
            case ButtonType.BARE:
            case ButtonType.SMALL_BARE:
                return Styles.color.transparent;
            default: 
                return Styles.color.gray.xx_dark;
        }
    }
}

const getPadding = (type: ButtonType) => {
    switch(type) {
        case ButtonType.SMALL_NORMAL: 
        case ButtonType.SMALL_GREEN:
        case ButtonType.SMALL_RED:
        case ButtonType.SMALL_BARE:
            return Styles.button.paddingSmall;
        case ButtonType.NORMAL: 
        case ButtonType.GREEN:  
        case ButtonType.RED:
        case ButtonType.BARE:
            return Styles.button.paddingNormal;
        default:
            return '10px'; 
    } 
    
}

const getTextColor = (props: StyledButtonProps): string => {
    if (props.disabled) {
        switch(props.buttonType) {
            case ButtonType.NORMAL:
            case ButtonType.SMALL_NORMAL:
            case ButtonType.GREEN:
            case ButtonType.SMALL_GREEN:
            case ButtonType.RED:
            case ButtonType.SMALL_RED:
                return Styles.color.white;
            case ButtonType.BARE:
            case ButtonType.SMALL_BARE:
                return Styles.color.gray.dark;
            default: 
                return Styles.color.white;
        }
    } else {
        switch(props.buttonType) {
            case ButtonType.NORMAL: 
            case ButtonType.SMALL_NORMAL:
            case ButtonType.GREEN: 
            case ButtonType.SMALL_GREEN:
            case ButtonType.RED:
            case ButtonType.SMALL_RED:
                return Styles.color.white;
            case ButtonType.BARE:
            case ButtonType.SMALL_BARE:
                return Styles.color.gray.xx_dark;
            default: 
                return Styles.color.gray.xx_dark;
        }
    }
}

const getShadow = (props: StyledButtonProps): string => {
    if (props.buttonType === ButtonType.BARE || 
        props.buttonType === ButtonType.SMALL_BARE || 
        props.disabled) 
    { 
        return '0'; 
    }
    else return Styles.button.shadow;

}

// eventually this implementation should be changed to something more sustainable
const Container = styled.button`
    border: 0;
    border-radius: 4px;
    background-color: ${(props: StyledButtonProps) => getBgColor(props)};
    color: ${(props: StyledButtonProps) => getTextColor(props)};
    box-shadow: ${(props: StyledButtonProps) => getShadow(props)};
    padding: ${(props: StyledButtonProps) => getPadding(props.buttonType)};
    :hover {
        cursor: ${(props: StyledButtonProps) => props.disabled ? 'default' : 'pointer'}
    }
    :focus {
        border: 0;
    }
`;

type ButtonProps = {
    buttonType?: ButtonType,
    text: string,
    styles?: object,
    onClick: () => void,
    disabled?: boolean,
}

export const Button: FunctionComponent<ButtonProps> = ({
    buttonType = ButtonType.NORMAL, 
    text,
    onClick,
    styles = {},
    disabled = false
}: ButtonProps) => {

    return (
        <Container 
            style={styles}
            onClick={onClick}
            disabled={disabled}
            buttonType={buttonType}
        >
            {text}
        </Container>
    );
}


export default Button;