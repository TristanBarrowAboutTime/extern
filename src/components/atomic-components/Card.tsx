import React, { ReactNode } from 'react';
import styled from 'styled-components';

type StyleProps = {
}

const Container = styled.div`
    display: flex;
    background-color: white;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
    box-sizing: border-box;
    border-radius: 4px;
    margin: 8px;
    padding: 16px;
    transition: all .2s ease;
    width: calc(100%/2 - 16px);

    @media only screen and (min-width: 1150px) {
        width: calc(100%/3 - 16px);    
    }

    @media only screen and (max-width: 850px) {
        width: auto;
        flex-grow: 1;
    }

    
`;

type CardProps = {
    children: ReactNode
}

const Card = (props: CardProps) => {
    return (
        <Container>
            {props.children}
        </Container>
    );
}

export default Card; 