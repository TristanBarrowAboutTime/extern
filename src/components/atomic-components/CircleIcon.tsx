import React from 'react';
import { View, StyleSheet, TouchableWithoutFeedback } from 'react-native';
import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Styles from '../../style/WMStyles'

type CircleIconProps = {
    icon: IconProp 
    color?: string
    bgColor?: string
    size?: number
    marginLeft?: number
    marginRight?: number
    circleSize?: number
    onPress?: () => void
}

const CircleIcon = ({
    icon,
    color = Styles.color.white,
    bgColor = Styles.color.gray.x_dark,
    size = 11,
    circleSize = 16,
    marginLeft = 0,
    marginRight = 0,
    onPress
}: CircleIconProps) => {
    const styles = StyleSheet.create({
        iconContainer: {
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            height: circleSize,
            width: circleSize,
            borderRadius: 100/2,
            backgroundColor: bgColor,
            fontSize: size,
            marginLeft: marginLeft,
            marginRight: marginRight,
            marginTop: 0,
            marginBottom: 0,
            '&:hover': {
                cursor: 'pointer'
            }
        }
});
    return (
        <TouchableWithoutFeedback onPress={onPress}>
            <View style={styles.iconContainer}>
                <FontAwesomeIcon icon={icon} color={color} />
            </View>
        </TouchableWithoutFeedback>
    );
} 



export default CircleIcon;