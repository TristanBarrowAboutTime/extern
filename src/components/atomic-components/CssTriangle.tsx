import React from 'react';
import styled from 'styled-components';

type StyleProps = {
    size: number,
    color: string,
}

const defaultColor = '#000000';
const defaultSize = 5;

const UpArrow = styled.div`
    width: 0;
    height: 0;
    border-left: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-right: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-bottom: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

const DownArrow = styled.div`
    width: 0;
    height: 0;
    border-left: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-right: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-top: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

const LeftArrow = styled.div`
    width: 0;
    height: 0;
    border-top: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-bottom: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-right: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

const RightArrow = styled.div`
    width: 0;
    height: 0;
    border-top: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-bottom: ${(props: StyleProps) => `${props.size}px solid transparent`};
    border-left: ${(props: StyleProps) => `${props.size}px solid ${props.color}`};
`;

type CssTriangleProps = {
    className?: string
    onClick?: () => void
    color?: string
    size?: number
}

export const Up = ({
    className = '',
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div className={className} onClick={onClick}>
            <UpArrow size={size} color={color} />
        </div>
    );
}

export const Down = ({
    className = '',
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div className={className} onClick={onClick}>
            <DownArrow size={size} color={color} />
        </div>
    );
}

export const Left = ({
    className = '',
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div className={className} onClick={onClick}>
            <LeftArrow size={size} color={color} />
        </div>
    );
}

export const Right = ({
    className = '',
    onClick,
    color = defaultColor,
    size = defaultSize,
}: CssTriangleProps) => {
    return (
        <div className={className} onClick={onClick}>
            <RightArrow size={size} color={color} />
        </div>
    );
}