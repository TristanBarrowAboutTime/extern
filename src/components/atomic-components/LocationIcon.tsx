import * as React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMapMarker } from '@fortawesome/free-solid-svg-icons';
import styled from 'styled-components';
import { useInitials } from '../../hooks/useInitials';

const Container = styled.div`
    position: relative;
`; 

type InitialsStyle = {
    size: number
    color: string
}

const Initials = styled.div`
    position: absolute;
    text-align: center;
    box-sizing: border-box;
    height: ${(props: InitialsStyle) => props.size}px;
    width: ${(props: InitialsStyle) => props.size * 0.75}px;
    font-size: ${(props: InitialsStyle) => props.size * 0.4}px;
    padding-top: ${(props: InitialsStyle) => props.size * 0.1}px;
    color: ${(props: InitialsStyle) => props.color};
    
`;


type LocationIconProps = {
    first: string
    last: string
    color?: string
    textColor?: string
    size?: number
}

const LocationIcon = (props: LocationIconProps) => {
    const { first, last, color, size, textColor } = props;

    const initals = useInitials({ first, last })
    return (
        <Container>
            <Initials 
                size={size as number} 
                color={textColor as string}
            >
                {initals}
            </Initials>
            <FontAwesomeIcon
                icon={faMapMarker} 
                color={color} 
                style={{fontSize: size}} 
            />
        </Container>

    ); 
}

LocationIcon.defaultProps = {
    color: 'gray',
    textColor: '#E9E9E9',
    size: 50,

}

export default LocationIcon;