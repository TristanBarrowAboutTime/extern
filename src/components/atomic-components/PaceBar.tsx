import React from 'react';
import { useMemo } from 'react'; 
import styled from 'styled-components';

type ProgressStyleProps = {
    width: string
    color: string
}

const FullBar = styled.div`
    position: relative;
    box-sizing: border-box;
    width: 100%;
    height: 24px;
    border: 1px solid black;
`;

const Progress = styled.div`
    background-color: ${(props: ProgressStyleProps) => props.color};
    width: ${(props: ProgressStyleProps) => props.width};
    height: 22px;
`;

const CenterMeter = styled.div`
    position: absolute;
    top: 0;
    left: 50%;
    width: 0;
    height: 20px;
    border: 1px solid black;
`;

type ProgressBarProps = {
    pace: number
    needed: number
}

const ProgressBar = ({pace, needed}: ProgressBarProps) => {
    const binding = usePaceBar({pace, needed});
    return (
        <FullBar>
            <Progress color={binding.color} width={binding.width} />
            <CenterMeter />
        </FullBar>
    );
}

type UsePaceBarArgs = {
    pace: number
    needed: number
}

const usePaceBar = (args: UsePaceBarArgs) => {

    const percent = useMemo(() => {
        let pace = args.pace;
        if (args.pace > args.needed) {
            pace = pace - args.needed;
            pace = 2 * ((1 / (1 + Math.pow(Math.E, -pace / args.needed))) - 0.5); // sigmoid
            pace = pace * args.needed + args.needed;
        }
        return pace / args.needed * 50;
    }, [args.pace, args.needed]);

    const color = useMemo(() => percent >= 50 ? 'green' : '', [percent]);
    const width = useMemo(() => `${percent}%`, [percent]);
    return {
        color,
        width
    }
}

export default ProgressBar;