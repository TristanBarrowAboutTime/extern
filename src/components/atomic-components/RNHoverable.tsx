import * as React from 'react';
import { Platform, TouchableOpacity } from 'react-native';
import styled from 'styled-components';

const WebContainer = styled.div`
    position: relative;

`;

type Position = {
    horizontal: number
    vertical: number
}

const Popout = styled.div`
    position: absolute;
    top: ${(props: { pos: Position }) => props.pos.vertical}px;
    left: ${(props: { pos: Position }) => props.pos.horizontal}px;
`;


type RNHoverableProps = {
    isInitToOpen?: boolean
    popout: React.ReactNode
    children: React.ReactChild | React.ReactChildren
    position: { horizontal: number, vertical: number }
}

export const RNHoverableWeb = ({
    isInitToOpen = false,
    popout,
    children,
    position
}: RNHoverableProps) => {
    const [isHovered, setIsHoveredTo] = React.useState(isInitToOpen);

    const onMouseEnter = () => setIsHoveredTo(true);
    const onMouseLeave = () => setIsHoveredTo(false);

    return (
        <>
            <WebContainer
                onMouseEnter={onMouseEnter}
                onMouseLeave={onMouseLeave}
            >
                {children}
                {isHovered && (
                    <Popout pos={position}>
                        {popout}
                    </Popout>
                )}
            </WebContainer>
        </>
    );
}

export const RNHoverableNative = ({
    isInitToOpen = false,
    popout,
    children
}: RNHoverableProps) => {
    const [isOpen, setIsOpenTo] = React.useState(isInitToOpen);

    const toggle = () => setIsOpenTo(!isOpen);

    return (
        <>
            <TouchableOpacity onPress={toggle}>
                {children}
            </TouchableOpacity>
            {isOpen && popout}
        </>
    );
}

export default Platform.select({
    native: RNHoverableNative,
    default: RNHoverableWeb 
});