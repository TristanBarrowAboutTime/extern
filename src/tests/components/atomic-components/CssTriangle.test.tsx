import React from 'react';
import renderer from 'react-test-renderer';
import { Up, Right, Down, Left } from '../../../components/atomic-components/CssTriangle';
import 'jest-styled-components';
import { unmountComponentAtNode } from 'react-dom';

let container: Element | DocumentFragment | null  = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container as Element | DocumentFragment);
    container = container as Element;
    if (container !== null) container.remove();
    container = null;

});

describe('CssTriangle Up', () => {
    it('gets clicked', () => {
        const mockFn = jest.fn();
        const component = renderer.create(
            <Up onClick={mockFn} />
        );
        component.root.findAllByType('div')[0].props.onClick();
        expect(mockFn).toHaveBeenCalled();
    });

    it('to match snapshot with no parameters', () => {
        const component = renderer.create(<Up />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with color modified', () => {
        const component = renderer.create(<Up color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with size modified', () => {
        const component = renderer.create(<Up size={10} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with size and color modified', () => {
        const component = renderer.create(<Up size={10} color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });

});

describe('CssTriangle Right', () => {
    it('gets clicked', () => {
        const mockFn = jest.fn();
        const component = renderer.create(
            <Right onClick={mockFn} />
        );
        component.root.findAllByType('div')[0].props.onClick();
        expect(mockFn).toHaveBeenCalled();
    });

    it('to match snapshot with no parameters', () => {
        const component = renderer.create(<Right />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with color modified', () => {
        const component = renderer.create(<Right color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });
    
    it('to match snapshot with size modified', () => {
        const component = renderer.create(<Right size={10} />).toJSON();
        expect(component).toMatchSnapshot();
    });
    
    it('to match snapshot with size and color modified', () => {
        const component = renderer.create(<Right size={10} color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });

});

describe('CssTriangle Down', () => {
    it('gets clicked', () => {
        const mockFn = jest.fn();
        const component = renderer.create(
            <Down onClick={mockFn} />
        );
        component.root.findAllByType('div')[0].props.onClick();
        expect(mockFn).toHaveBeenCalled();
    });

    it('to match snapshot with no parameters', () => {
        const component = renderer.create(<Down />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with color modified', () => {
        const component = renderer.create(<Down color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with size modified', () => {
        const component = renderer.create(<Down size={10} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with size and color modified', () => {
        const component = renderer.create(<Down size={10} color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });

});

describe('CssTriangle Left', () => {
    it('gets clicked', () => {
        const mockFn = jest.fn();
        const component = renderer.create(
            <Left onClick={mockFn} />
        );
        component.root.findAllByType('div')[0].props.onClick();
        expect(mockFn).toHaveBeenCalled();
    });

    it('to match snapshot with no parameters', () => {
        const component = renderer.create(<Left />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with color modified', () => {
        const component = renderer.create(<Left color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with size modified', () => {
        const component = renderer.create(<Left size={10} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('to match snapshot with size and color modified', () => {
        const component = renderer.create(<Left size={10} color={'#abcdef'} />).toJSON();
        expect(component).toMatchSnapshot();
    });

});