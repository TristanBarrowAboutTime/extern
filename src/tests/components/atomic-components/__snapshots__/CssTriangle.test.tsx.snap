// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CssTriangle Down to match snapshot with color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={5}
  />
</div>
`;

exports[`CssTriangle Down to match snapshot with no parameters 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={5}
  />
</div>
`;

exports[`CssTriangle Down to match snapshot with size and color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-top: 10px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={10}
  />
</div>
`;

exports[`CssTriangle Down to match snapshot with size modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-top: 10px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={10}
  />
</div>
`;

exports[`CssTriangle Left to match snapshot with color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
  border-right: 5px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={5}
  />
</div>
`;

exports[`CssTriangle Left to match snapshot with no parameters 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
  border-right: 5px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={5}
  />
</div>
`;

exports[`CssTriangle Left to match snapshot with size and color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 10px solid transparent;
  border-bottom: 10px solid transparent;
  border-right: 10px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={10}
  />
</div>
`;

exports[`CssTriangle Left to match snapshot with size modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 10px solid transparent;
  border-bottom: 10px solid transparent;
  border-right: 10px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={10}
  />
</div>
`;

exports[`CssTriangle Right to match snapshot with color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
  border-left: 5px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={5}
  />
</div>
`;

exports[`CssTriangle Right to match snapshot with no parameters 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
  border-left: 5px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={5}
  />
</div>
`;

exports[`CssTriangle Right to match snapshot with size and color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 10px solid transparent;
  border-bottom: 10px solid transparent;
  border-left: 10px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={10}
  />
</div>
`;

exports[`CssTriangle Right to match snapshot with size modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-top: 10px solid transparent;
  border-bottom: 10px solid transparent;
  border-left: 10px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={10}
  />
</div>
`;

exports[`CssTriangle Up to match snapshot with color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 5px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={5}
  />
</div>
`;

exports[`CssTriangle Up to match snapshot with no parameters 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 5px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={5}
  />
</div>
`;

exports[`CssTriangle Up to match snapshot with size and color modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-bottom: 10px solid #abcdef;
}

<div
  className=""
>
  <div
    className="c0"
    color="#abcdef"
    size={10}
  />
</div>
`;

exports[`CssTriangle Up to match snapshot with size modified 1`] = `
.c0 {
  width: 0;
  height: 0;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-bottom: 10px solid #000000;
}

<div
  className=""
>
  <div
    className="c0"
    color="#000000"
    size={10}
  />
</div>
`;
